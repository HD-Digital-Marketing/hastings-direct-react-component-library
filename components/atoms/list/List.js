import React from 'react'
import classNames from 'classnames'
import PropTypes from 'prop-types'
import './list.scss'

const List = props => {

    const {
        listItems = [],
        listType = 'ul',
        isBold = false,
        theme = '',
        className = ''
    } = props

    const listClass = classNames('a-list', className, {
        'a-list--numbered': listType === 'ol',
        'a-list--dark': theme === 'dark',
        'a-list--primary': theme === 'primary',
        'a-list--primary a-list--primary-on-dark': theme === 'primary-dark',
        'a-list--secondary': theme === 'secondary',
        'a-list--secondary a-list--secondary-on-dark': theme === 'secondary-dark',
        'a-list--tick a-list--tick-primary': theme === 'tick-primary',
        'a-list--tick a-list--tick-primary a-list--tick-primary-on-dark': theme === 'tick-primary-dark',
        'a-list--tick a-list--tick-secondary': theme === 'tick-secondary',
        'a-list--tick a-list--tick-secondary a-list--tick-secondary-on-dark': theme === 'tick-secondary-dark',
        'a-list--bold': isBold
    })

    const renderUnorderedList = data => {
        return (
            <ul className={listClass}>
                {data.map((value, index) =>
                    <li key={index}>{value}</li>
                )}
            </ul>
        )
    }

    const renderOrderedList = data => {
        return (
            <ol className={listClass}>
                {data.map((value, index) =>
                    <li key={index}>{value}</li>
                )}
            </ol>
        )
    }

    return(
        listType === 'ol' ? renderOrderedList(listItems) : renderUnorderedList(listItems)
    )
}

List.propTypes = {
    listItems: PropTypes.array,
    listType: PropTypes.string,
    isBold: PropTypes.bool,
    theme: PropTypes.string,
    className: PropTypes.string
}

export default List
