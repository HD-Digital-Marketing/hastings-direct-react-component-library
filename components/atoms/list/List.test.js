import React from 'react'
import renderer from 'react-test-renderer';
import List from './List'

describe('Check the list renders', () => {
    test('it should match the snapshot', () => {

        const listItems = [
            "Car policy",
            "Home policy",
            "Bike policy",
            "Van policy",
            "Multicar policy",
            "YouDrive policy"
        ]

        const tree = renderer
            .create(<List theme="tick-secondary-dark" isBold={true} listItems={listItems} />)
            .toJSON()
        expect(tree).toMatchSnapshot()

    })
})