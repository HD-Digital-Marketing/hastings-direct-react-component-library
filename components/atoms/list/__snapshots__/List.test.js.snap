// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check the list renders it should match the snapshot 1`] = `
<ul
  className="a-list a-list--tick a-list--tick-secondary a-list--tick-secondary-on-dark a-list--bold"
>
  <li>
    Car policy
  </li>
  <li>
    Home policy
  </li>
  <li>
    Bike policy
  </li>
  <li>
    Van policy
  </li>
  <li>
    Multicar policy
  </li>
  <li>
    YouDrive policy
  </li>
</ul>
`;
